From f30f9c69f13df163ca127ddc16362d71037d6331 Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Mon, 6 Feb 2012 13:17:11 +0900
Subject: [PATCH 07/73] mach-rcar: fix meram size to access through UIO

In R-CarE1, just 128 KiB MERAM could be mounted, furthermore
the last 32KiB region must be reserved in the default H/W configuration.
This patch fixes the range of MERAM to access through UIO.
---
 arch/arm/mach-rcar/board-silverstone.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-rcar/board-silverstone.c b/arch/arm/mach-rcar/board-silverstone.c
index 6fb554b..f56dfdc 100644
--- a/arch/arm/mach-rcar/board-silverstone.c
+++ b/arch/arm/mach-rcar/board-silverstone.c
@@ -96,7 +96,7 @@ static struct resource meram_resources[] = {
 	[1] = {
 		.name   = "MERAM_MEM",
 		.start  = 0xfd480000,
-		.end    = 0xfd4fffff,
+		.end    = 0xfd480000 + ((128 - 32) << 10) - 1,	/* 32KiB reserved for cache */
 		.flags  = IORESOURCE_MEM,
 	}
 };
-- 
1.7.0.4

